src/olympia/bandwagon/serializers.py

```diff
@@ -109,3 +109,18 @@ def validate(self, data):
             # DRF normally ignores updates to read_only fields, so do the same.
             data.pop('addon')
         return super(CollectionAddonSerializer, self).validate(data)
+
+
+class CollectionWithAddonsSerializer(CollectionSerializer):
+    addons = serializers.SerializerMethodField()
+
+    class Meta(CollectionSerializer.Meta):
+        fields = CollectionSerializer.Meta.fields + ('addons',)
+        read_only_fields = tuple(
+            set(fields) - set(CollectionSerializer.Meta.writeable_fields))
+
+    def get_addons(self, obj):
+        addons_qs = self.context['view'].get_addons_queryset()
+        return [
+            CollectionAddonSerializer(addon).data for addon in addons_qs
```

Comment:
You should be able to just do `return CollectionAddonSerializer(addons_qs, many=True).data` instead.

In fact, it may even work with `addons = CollectionAddonSerializer(source='fetch_addons', many=True)` above and just `return self.context['view'].get_addons_queryset()` in a `fetch_addons` method. 

Any of those 2 approaches should work (hopefully, haven't tested) and avoids creating extra serializer instances, which is a bit costly.

Reply:
latter didn't work bit did the former https://github.com/mozilla/addons-server/pull/7546/commits/02830c9e8bcbb34e725ea106f697b18951f2f730